components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIDefinitionFromSwagger20.java

```diff
@@ -412,6 +422,16 @@ public String generateSwaggerFromResources(API.APIBuilder api) {
                 operation.setConsumes(consumesList);
             }
             operation.addResponse("200", getDefaultResponse());
+            if (!uriTemplate.getAuthType().equals(APIMgtConstants.AUTH_NO_AUTHENTICATION) &&
+                    ((api.getSecurityScheme() & 2) == 2)) {
+              operation.addSecurity("apikey", null);
+            }
+
+            if (!uriTemplate.getAuthType().equals(APIMgtConstants.AUTH_NO_AUTHENTICATION) &&
+                    ((api.getSecurityScheme() & 1) == 1)) {
+                operation.addSecurity("oauth2", null);
+            }
+
```

Comment:
'apikey', 'oauth2' are hardcoded in multiple places, please refactor to a common constant.

Reply:
will do